### DISCUSSION TITLE
[v3] $this->forgetComputed

### DISCUSSION THREAD

**Discussion by julesgraus** [livewire/livewire#5905] [1. Help] [answered]
In Livewire 2 you could use ```$this->forgetComputed``` inside of your components. Is it gone in version 3? It isn't mentioned in the upgrade guide.

**Comment by marksalmon** [comment, marked as answer]
In V3 you have to use php's ```unset()``` function.

See [here in the docs](https://livewire.laravel.com/docs/computed-properties#busting-the-cache)

**Comment by julesgraus** [comment]
Thanks!

The upgrade guide still doesnt mention this. And the upgrade tool doesnt handle it too. Is it okay if i update upgrade guide if i find some time? I dont know how to "update" the upgrade tool, else i would like to do that too if someone gives me a direction on that.

**Reply by joshhanley** [reply]
@julesgraus yep, PRs are welcome! @PhiloNL is our man for the upgrade tool he might be able to provide guidance. 🙂

**Reply by julesgraus** [reply]
@PhiloNL I think I know what I need to do. I will create a PR when I am ready. May take me an evening or 2. Dankjewel

**Reply by julesgraus** [reply]
Update: I've created an upgrade command using Philo's logic that updates forgetComputed calls to unset()'s. Now I am creating a command that will upgrade computed properties from the old syntax to the new one with the computed attribute above it. For this I cannot simply uses the existing logic because it's a little bit more complex. I still will take an attempt in doing so. Will send an update tomorrow around this time.

**Reply by joshhanley** [reply]
@julesgraus thanks! Can you submit separate PRs for the changes, one for the forgetComputed and another for the computed change (when done)?

**Reply by julesgraus** [reply]
Sure, will do!

**Reply by joshhanley** [reply]
@julesgraus sweet thanks! 🙂

**Comment by julesgraus** [comment]
The first PR is ready. See https://github.com/livewire/livewire/pull/5956

The other one will come within 9 hours. I do have it pretty much ready.

**Reply by julesgraus** [reply]
The second one also is ready. Please have a look at it over here: https://github.com/livewire/livewire/pull/5971

**Reply by julesgraus** [reply]
Took some screenshots of the results of the second PR. For what its worth ;)
<img width="1159" alt="command" src="https://github.com/livewire/livewire/assets/4387442/0f95a2a4-5044-449b-bb2a-e59d01abcfe2">
<img width="1112" alt="Import result" src="https://github.com/livewire/livewire/assets/4387442/082dd3cf-ca89-4bd9-93f4-6eca08d9d302">
<img width="924" alt="abstract function upgrade result" src="https://github.com/livewire/livewire/assets/4387442/1a492d09-ef97-4c87-aa98-c3bf39909b3b">
<img width="597" alt="function upgrade result" src="https://github.com/livewire/livewire/assets/4387442/3a18b683-c51b-4cf7-952f-e4e7f25967e6">

**Reply by joshhanley** [reply]
@julesgraus thanks!
